tests/unit/test_fp16.py

```diff
@@ -8,17 +8,51 @@
 from simple_model import SimpleModel, SimpleOptimizer, random_dataloader, args_from_dict
 
 
+def test_lamb_fp32_grad_clip(tmpdir):
+    config_dict = {
+        "train_batch_size": 2,
+        "steps_per_print": 1,
+        "optimizer": {
+            "type": "Lamb",
+            "params": {
+                "lr": 0.00015
+            }
+        },
+        "gradient_clipping": 1.0
+    }
+    args = args_from_dict(tmpdir, config_dict)
+    hidden_dim = 10
+
+    model = SimpleModel(hidden_dim, empty_grad=False)
+
+    @distributed_test(world_size=[1, 2])
+    def _test_lamb_fp16_basic(args, model, hidden_dim):
+        model, _, _,_ = deepspeed.initialize(args=args,
+                                             model=model,
+                                             model_parameters=model.parameters())
+        data_loader = random_dataloader(model=model,
+                                        total_samples=50,
+                                        hidden_dim=hidden_dim,
+                                        device=model.device)
+        for n, batch in enumerate(data_loader):
+            loss = model(batch[0].float(), batch[1])
+            model.backward(loss)
+            model.step()
+
```

Comment:
Dont we need to compare the loss with a reference? or is the test to just ensure that the parameters can be passed?

Reply:
That sort of test is done in our model tests, this will just test to make sure everything is train-able. So should be fine :) Would be great to add loss checking in our unit tests though.